packages/backend-plugin-api/api-report.md

```diff
@@ -229,6 +293,30 @@ export interface PluginMetadataService {
   getId(): string;
 }
 
+// @public (undocumented)
+export interface PluginServiceFactoryConfig<
+  TService,
+  TContext,
+  TImpl extends TService,
+  TDeps extends {
+    [name in string]: ServiceRef<unknown>;
+  },
+> {
+  // (undocumented)
+  createRootContext?(
+    deps: ServiceRefsToInstances<TDeps, 'root'>,
+  ): Promise<TContext>;
```

Comment:
Honestly now that we've simplified the structure of this interface (as in, the formerly more nested return types), I think we should at the same time make them return `T | Promise<T>`. Same for the factory itself. I imagine especially for the context maker, that it'll almost never _want_ to be async. It'll be a nice convenience, at little-to-no complexity cost for the framework

Reply:
Yeah I saw this pattern when refactoring and thought the same. Maybe it's worth supporting.